mixer/adapter/svcctrl/handler.go

```diff
@@ -97,39 +102,76 @@ func newServiceProcessor(meshServiceName string, ctx *handlerContext) (*serviceP
 
 // HandleApiKey handles apikey check.
 func (h *handler) HandleApiKey(ctx context.Context, instance *apikey.Instance) (adapter.CheckResult, error) {
-	result, err := h.svcProc.ProcessCheck(ctx, instance)
-	logger := h.ctx.env.Logger()
+	svcProc, err := h.getServiceProcessor(ctx)
 	if err != nil {
-		logger.Errorf("svcctrl check failed: %v", err)
+		return adapter.CheckResult{
+			Status: status.WithPermissionDenied(err.Error()),
+		}, nil
 	}
-	return result, err
+	return svcProc.ProcessCheck(ctx, instance)
 }
 
 // HandleSvcctrlReport handles reporting metrics and logs.
 func (h *handler) HandleSvcctrlReport(ctx context.Context, instances []*svcctrlreport.Instance) error {
-	err := h.svcProc.ProcessReport(ctx, instances)
-	logger := h.ctx.env.Logger()
+	svcProc, err := h.getServiceProcessor(ctx)
 	if err != nil {
-		logger.Errorf("svcctrl check failed: %v", err)
+		return err
 	}
-	return err
+	return svcProc.ProcessReport(ctx, instances)
 }
 
 // HandleQuota handles rate limiting quota.
 func (h *handler) HandleQuota(ctx context.Context, instance *quota.Instance,
 	args adapter.QuotaArgs) (adapter.QuotaResult, error) {
-	return h.svcProc.ProcessQuota(ctx, instance, args)
+	svcProc, err := h.getServiceProcessor(ctx)
+	if err != nil {
+		return adapter.QuotaResult{
+			Status: status.WithPermissionDenied(err.Error()),
```

Comment:
Is this internal error? It does not look like a permission issue.

Reply:
fixed